packages/squiggle-lang/src/rescript/utility/Sparklines.res

```diff
@@ -0,0 +1,40 @@
+// Port of Sindre Sorhus' Sparkly to Rescript
+// reference implementation: https://github.com/sindresorhus/sparkly
+// Omitting rgb "fire" style, so no `chalk` dependency
+
+type sparklyConfig = {
+  minimum: option<float>,
+  maximum: option<float>
+}
+
+let sparkly = (
+  numbers: array<float>,
+  ~options = {minimum: None, maximum: None}
+) => {
+  // Unlike reference impl, we assume that all numbers are finite, i.e. no NaN.
+
+  let ticks = [`▁`, `▂`, `▃`, `▄`, `▅`, `▆`, `▇`, `█`]
+
+  let minimum = E.O.default(Js.Math.minMany_float(numbers), options.minimum)
+  let maximum = E.O.default(Js.Math.maxMany_float(numbers), options.maximum)
+
+  // Use a high tick if data is constant and max is not equal
+  let ticks = if minimum == maximum && maximum != 0.0 {
+    [ticks[4]]
```

Comment:
Now that I think about it, I think this should return an error instead of just a lot of high values.

Reply:
That seems good. Maybe `a could just be a string at this point.

Reply:
I don't want to do `result` type for now. I'm not being paranoid about breaking cases, like `NaN`, for this pass. but moreover this specific question of constant data I think shouldn't artificially inflate it should just show the flat line wherever it falls (usually all across the bottom, depending on the distance between the constant and 1)

Reply:
Do you think it should be robust to cases like `NaN` on this pass through? @OAGr

Reply:
It shouldn't ever get NaN, as it's Rescript, unless we're doing something weird.